src/Recurr/Transformer/InstanceGenerator.php

```diff
@@ -0,0 +1,770 @@
+<?php
+
+/*
+ * Copyright 2013 Shaun Simmons
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * Based on:
+ * rrule.js - Library for working with recurrence rules for calendar dates.
+ * Copyright 2010, Jakub Roztocil and Lars Schoning
+ * https://github.com/jkbr/rrule/blob/master/LICENCE
+ */
+
+namespace Recurr\Transformer;
+
+use DateTime;
+use DateTimeZone;
+use Generator;
+use Recurr\DateUtil;
+use Recurr\Exception\MissingData;
+use Recurr\Frequency;
+use Recurr\RecurrenceCollection;
+use Recurr\Rule;
+use Recurr\Time;
+use Recurr\Weekday;
+
+/**
+ * Expands instances for a given RRULE.
+ *
+ * @package Recurr
+ */
+class InstanceGenerator
+{
+    /**
+     * Some versions of PHP are affected by a bug where
+     * DateTime::createFromFormat('z Y', ...) does not account for leap years.
+     *
+     * @var bool
+     */
+    protected $leapBug = false;
+
+    /**
+     * Constructor
+     */
+    public function __construct()
+    {
+        $this->leapBug = DateUtil::hasLeapYearBug();
+    }
+
+    /**
+     * Returns a generator that yields instances for the given rules and dates.
+     *
+     * @param DateTime $seed The seed date that serves as the starting point to recur from. Instances will be
+     *   expanded within the timezone set in this seed.
+     * @param Rule[] $rrules A collection of iCal RRULE values to expand by.
+     * @param DateTime[] $rdates A collection of iCal RDATE values to include in the expansion.
+     * @param Rule[] $exrules A collection of iCal EXRULE values to expand and exclude.
+     * @param DateTime[] $exdates A collection of iCal EXDATE values to exclude in the expansion. (NOTE: EXDATE was
+     *   introduced in RFC2445 and deprecated in RFC5545. It is included here for legacy support because it is commonly
+     *   supported by popular cloud-based calendar systems like Google Calendar.
+     * @param bool $ignoreCounts Ignores the `COUNT` constraint of all rules given. This is useful if the `$seed`
+     *   represents a date in the middle of the series and using `COUNT` would be cause for an inaccurate end date.
+     * @param int|null $iterationLimit An iteration limit that, when provided, will serve as a fail-safe
+     *   that stops the generator when the limit is reached to prevent an infinitely large iterator.
+     * @return Generator|DateTime[]
```

Comment:
It's not that I don't understand that. The alternative is a more truthy docblock and less convenient integration with tooling. I'd argue the tradeoff is better with this as is, since it does behave like an array of `DateTime` you can use it this way pretty safely. 

There has been discussion on a `@yield` tag, but no traction or tooling support (that I know of). Similarly there has been a `Generator<DateTime, ...>` syntax suggested for the PSR, but no traction on that in over a year.

This is just a little pragmatic.